### PR TITLE
added description for private attr_accessor

### PR DESCRIPTION
**Pull request by sullysaurus**
@jmburges @SophieDeBenedetto @AnnJohn

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,7 +123,13 @@ bub.age
 NoMethodError: private method `age' called for #<Cat:0x000001020bfce0 @name="Lil Bub", @age=2>
 ```
 
-Let's make a method in the `Cat` class that implicitly receives age, through the only way it can: `self`. A cat can have a birthday, where it ages 1 year when the birthday happens. 
+### Why Have A Private `attr_accessor`?
+
+Previously we were using getters/setters so the outside world could access our instance variables. But now we are using them so that we can locally access them. Why is that valuable? A lot of software bugs come from the incorrect modification of state, aka variables. When we are debugging our program, because so many bugs come from modification of variables, if we can centralize all modification into one place we help with debugging. That's the beauty of enforcing getters/setters. Want to see all the places an instance variable is modified? With a private `attr_accessor` you can just override it and put a `binding.pry`.
```
Comment (jmburges) [inline README.md]: I wouldn't just copy paste what I write. These are pretty stream of consciousnesses things. There are definitely too many commas in my setnence that starts with "When we are debugging.." Also add an example of overriding the setter with a pry. Also "With a private `attr_Accessor` you can just override it" I don't think will make sense to people. When I say override it, what do I mean? I mean the setter but I need you to sorta translate that to what the student will identify ith

File: README.md [hunk 2]
```diff
@@ -123,7 +123,13 @@ bub.age
...
+
+Another nice feature of using setters/getters is future proofing. Maybe in the future you want to add validations whenever someone sets one of your instance variables? No problem! Just override the setter method. No need to find all the places that you modify that instance variable and change it.
+
+Let's make a method in the `Cat` class that implicitly receives age, through the only way it can: `self`. A cat can have a birthday, where it ages 1 year when the birthday happens.
```
Comment (jmburges) [inline README.md]: remove "implicitly receives age" it's not implicit at all. It's explicitly defined in the method defiition

File: README.md [hunk 3]
```diff
@@ -158,4 +164,3 @@ end
 Then when we call `bub.birthday!`, Lil Bub's age becomes 3. Happy birthday, Bub!
 
 ![bub birthday](http://ihavecat.com/wp-content/uploads/2014/06/480ae292-8e53-41ff-9b5a-868ef5f75d42.jpg)
```
Comment (jmburges) [inline README.md]: we don't want to hotlink to another persons images. What if they change their links or their website goes down? Use [this guide](https://github.com/learn-co-curriculum/learn-readme-style-guide#amazon-s3-1) on how to upload images to s3

**Comment by jmburges** [comment]
Don't forget to tag it against the issue!

Fixes Issue #5

File: README.md (hunk 2 above)
Comment (sullysaurus) [inline README.md]: Updated.

File: README.md (hunk 1 above)
Comment (sullysaurus) [inline README.md]: Updated.

File: README.md (hunk 3 above)
Comment (sullysaurus) [inline README.md]: Updated.

**Comment by jmburges** [comment]
:shipit:

**Pull request merged**